hknweb/exams/models.py

```diff
@@ -1,30 +1,39 @@
 from django.db import models
 
 class Department(models.Model):
-    name = models.CharField(unique=True, max_length=4, null=True) #short form (e.g. 'EE')
-    subject = models.CharField(max_length=255) #long form (e.g. 'Electrical Engineering')
+    abbreviated_name = models.CharField(unique=True, max_length=4, null=True) #short form (e.g. 'EE')
+    long_name = models.CharField(max_length=255) #long form (e.g. 'Electrical Engineering')
+
+    def getAbbrName(self):
+        return self.abbreviated_name
 
     def __str__(self):
-        return "{}".format(self.subject)
+        return "{} ({})".format(self.long_name, self.abbreviated_name)
 
 class Instructor(models.Model):
     name = models.CharField(max_length=255)
     department = models.ForeignKey(Department, on_delete=models.CASCADE, null=False)
 
     def __str__(self):
-        return "Instructor(name={}, department={})".format(self.name, self.department)
+        return "{}".format(self.name) #"Instructor(name={}, department={})".format(self.name, self.department)
 
 class Course(models.Model):
     name        = models.CharField(max_length=255, null=False)
     department  = models.ForeignKey(Department, on_delete=models.CASCADE, null=False)
     number 		= models.CharField(max_length=10, null=False)
 
     def __str__(self):
-        return "{} {}".format(self.department, self.number)
+        return "{} {}".format(self.department.getAbbrName(), self.number)
+
+class Semester(models.Model):
+     semester = models.CharField(max_length=255)
+
+     def __str__(self):
+         return "{}".format(self.semester)
 
 class CourseSemester(models.Model):
     course      = models.ForeignKey(Course, on_delete=models.CASCADE, null=True)
-    semester    = models.CharField(max_length=255)
+    semester    = models.ForeignKey(Semester, on_delete=models.CASCADE)
```

Comment:
Ok question. If a semester gets deleted do we want to delete the exam with it? (from `models.CASCADE`) We might want something like PROTECT (which prevents the deletion).

Reply:
Current precedence in other parts of the HKN website uses CASCADE

We can use "SET_DEFAULT" if desired

Reply:
Other areas:
- Candidates
- Events
- (Main Website)

Reply:
Then no action will be done here